README.md

```diff
@@ -1,17 +1,224 @@
 
-This is a WIP collection of Swift libraries enabling Tracing for your Swift (Server) systems. Check out the [GSoC project overview](https://summerofcode.withgoogle.com/projects/#6092707967008768) to learn more. For a more detailed project plan please take a look [at this Google doc](https://docs.google.com/document/d/19j9x515dR0IAwF3Zoevxoj6jvMdGpP4UuyGhmEXO_B8).
+This is a collection of Swift libraries enabling the instrumentation of your server side applications using tools such
+as tracers. Our goal is to provide a common foundation that allows you too freely choose how to instrument your system
+with minimal changes to your actual code.
 
-## Dependencies
+---
 
-The Swift packages contained in this repository make use of the library package `Baggage`, which can be found in a separate repository: https://github.com/slashmo/gsoc-swift-baggage-context
+## Instrumentation
+
+When instrumenting server applications there are typically three parties involved:
+
+1. [End-Users](#end-users-setting-up-instruments) developing server-side applications
+2. [Library/Framework authors](#libraryframework-authors-instrumenting-your-software) providing building blocks to create these applications
+3. [Instrument developers](#instrument-developers-creating-an-instrument) providing tools to collect distributed metadata about your application
+
+For server-side applications to be instrumented correctly, i.e. providing useful information about your system, these
+three parts have to play along nicely.
+
+Let's say you build an API for a fruit store that has two services, one for ordering goods and one that checks what
+items are available in storage. Your frontend might make an HTTP request to the order service which then makes a
+subsequent request to the storage service, just to find out that your application isn't behaving not as it should. Now
+you could debug each service individually to figure out where the bug is located. If you'd instead had instrumentation
+in place you could take a look take at the bigger picture instead.
+
+One specific type of instrument that might be helpful in this case is called *Tracer*. It collects pieces of metadata
+about your system (spans) and then links them to present the entire call from start to finish (trace). This metadata
+contains, depending on the piece being instrumented, e.g. the request URL or an SQL query. The following image shows
+one such trace presented in [Zipkin](https://zipkin.io):
+
+![Screenshot of trace presented in Zipkin](images/zipkin_trace.png)
+
+**TODO**: Replace with fruit store example screenshot
+
+In the fruit store example, we'd want a span for both the incoming HTTP request to the order service, its outgoing
+request to the storage service, one for handling that request, and a span respresenting the querying of a database.
+Each of these spans would contain information useful for debugging the interplay of multiple spans.
```

Comment:
```suggestion

Spans naturally form a parent-child relationship, and by visually analyzing a trace we can easily spot bottlenecks for performance profiling use-cases, and for failure debugging situations we can benefit from the application specific metadata that spans can carry (such as noticing that failures only occur e.g. when the username contains some illegal character or some other otherwise hard to notice situation).
```